src/ert/gui/simulation/simulation_panel.py

```diff
@@ -219,6 +221,9 @@ def runSimulation(self):
 
             delete_runpath_checkbox = None
 
+            if is_restart_run:
+                model.set_start_iteration(self._notifier.current_ensemble.iteration + 1)
```

Comment:
Is there a way we can avoid `current_ensemble` altogether? Feels a bit fragile, and think it could quite easily introduce some strange bugs because ultimately it is the run model which decides what we start from. It is a bit of a corner case, but for example:

1. Run ES-MDA (creating default_%d)
2. Enter Experiment manager
3. Create a new experiment and an ensemble named: default_2
4. Select the newly created ensemble in ES-MDA and restart
5. You will restart from the existing first es-mda run, but `start_iteration` will be 1, though the run model will start from 3.

Reply:
Perhaps we should first make it possible to select an experiment in ES-MDA, so that only the currently selected experiment's ensembles are available in the "Restart from" drop-down?

Reply:
That might be a solution, we check the runpath before we start the experiment, so making the run model own the starting iteration means moving creating the experiment to the constructor instead of in `run_experiment`?